mgr/src/server/datastore/volumes.cr

```diff
@@ -281,4 +281,57 @@ module Datastore
     query = "UPDATE volumes SET name = ? WHERE pool_id = ? AND id = ?"
     connection.exec(query, new_volname, pool_id, volume_id)
   end
+
+  def volume_name_exists_by_pool_id?(name, pool_id)
+    query = "SELECT COUNT(name) FROM volumes WHERE name = ? and pool_id = ?"
+    connection.scalar(query, name, pool_id).as(Int64) > 0
+  end
+
+  def node_part_of_other_volume?(pool_id, vol_id)
+    query = "DROP table if exists t1"
```

Comment:
```
node_ids = `SELECT DISTINCT node_id FROM storage_units WHERE volume_id = ?`

node_ids_arg = node_ids.map do |node_id|
    "?"
end

`SELECT COUNT(id) FROM storage_units WHERE volume_id != ? AND pool_id = ? AND node_id IN (#{node_ids_arg.join(",")})`
```

First query gets the list of nodes from the Volume, if Datastore.get_volume is called then you can skip the first query and use node ids directly.

Second query returns the count of Storage units that are part of the above nodes. If count is greater than zero then Error.

Reply:
Example of map: https://github.com/kadalu/moana/blob/main/mgr/src/server/datastore/nodes.cr#L101-L102

Reply:
Ack, Will add tests, once issue #250 is fixed. Since tests has node deletion part.